src/lib.rs

```diff
@@ -192,6 +192,16 @@ mod parser;
 mod raw_bmp;
 mod raw_iter;
 
+macro_rules! propagate {
+    ($e:expr) => {
+        match $e {
+            Ok(ok) => ok,
+            Err(e) => return Err(e),
+        }
+    };
+}
+pub(crate) use propagate;
```

Comment:
I would suggest to rename this macro into `try_const`, because long time Rust users will still be familiar with the old `try` macro.

We should also add a doc comment to explain the reason this macro exists and a note to remove this later, like:
```
/// Alternative to the `?` operator that is usable in const contexts.
// MSRV: Replace macro with the `?` operator when it gets supported in const context.
```